### PR TITLE
Add node start command - Closes #686

### LINKED ISSUE #686: Add node:start command
### Description

`node:start` command implementation, where it starts the instance of installed package using `node:install`

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Adding Node start command.

### Review checklist

* The PR resolves #686 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: src/commands/node/install.ts [hunk 1]
```diff
@@ -147,13 +142,13 @@ export default class InstallCommand extends BaseCommand {
 		} = flags as Flags;
 		const cacheDir = this.config.cacheDir;
 
-		const tasks = new Listr.default([
+		const tasks = new Listr([
 			{
-				title: `Install Lisk Core ${network}`,
+				title: `Install Lisk Core ${network} as ${name}`,
```
Comment (ManuGowda) [inline src/commands/node/install.ts:153]: The message appears like `Install Lisk Core testnet as newtest` I am not sure if we want to go with this name, for now, it looks ok, but I would suggest we can have a solid set of good names for these messages as they will be key for every user in terms of presentation. Let's revisit all the messages part at the last, I will create an issue for that.

File: src/commands/node/install.ts [hunk 2]
```diff
@@ -147,13 +142,13 @@ export default class InstallCommand extends BaseCommand {
...
 				task: () =>
-					new Listr.default([
+					new Listr([
 						{
-							title: 'Build options',
+							title: 'Setup Build options',
```
Comment (ManuGowda) [inline src/commands/node/install.ts]: `Setup Install Options`?

File: src/commands/node/install.ts [hunk 3]
```diff
@@ -187,13 +182,7 @@ export default class InstallCommand extends BaseCommand {
 						},
 						{
 							title: 'Download blockchain snapshot',
```
Comment (ManuGowda) [inline src/commands/node/install.ts]: `Download Blockchain Snapshot`

File: src/commands/node/stop/cache.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+export default class CacheCommand extends BaseCommand {
+	static description = 'Stop Lisk Core Cache';
+
+	static examples = [
+		'node:stop:cache',
+		'node:stop:cache --network=testnet',
+		'node:stop:cache --installation-path=/opt/lisk/lisk-testnet --network=testnet',
+	];
+
+	static flags = {
```
Comment (ManuGowda) [inline src/commands/node/stop/cache.ts:43]: Fix me
```
static flags = {
		...BaseCommand.flags,
		network: flagParser.string({
			...commonFlags.network,
			default: NETWORK.MAINNET,
			options: [NETWORK.MAINNET, NETWORK.TESTNET, NETWORK.BETANET],
		}),
		installationPath: flagParser.string({
			...commonFlags.installationPath,
			default: defaultInstallationPath,
		}),
		name: flagParser.string({
			...commonFlags.name,
			default: NETWORK.MAINNET,
		}),
	};
```

File: src/commands/node/stop/database.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	static description = 'Stop Lisk Core Database';
+
+	static examples = [
+		'node:stop:database',
+		'node:stop:database --no-snapshot',
+		'node:stop:database --network=testnet',
+		'node:stop:database --installation-path=/opt/lisk/lisk-testnet --network=testnet',
+	];
+
+	static flags = {
```
Comment (ManuGowda) [inline src/commands/node/stop/database.ts:42]: same here, fix here as start database changes.

File: src/utils/constants.ts [hunk 6]
```diff
@@ -71,3 +71,6 @@ export enum OS {
 	Darwin = 'MACOS',
 	Linux = 'LINUX',
 }
+
+export const RELEASE_URL = 'https://downloads.lisk.io/lisk';
+export const SNAPSHOT_URL = 'http://snapshots.lisk.io.s3-eu-west-1.amazonaws.com/lisk';
```
Comment (ManuGowda) [inline src/utils/constants.ts]: fix newline at the end of file

File: src/utils/node/pm2.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
+import * as path from 'path';
+import { connect, describe, disconnect, list, ProcessDescription, restart, start, stop } from 'pm2';
```
Comment (ManuGowda) [inline src/utils/node/pm2.ts]: what's your take on promisifying the whole `pm2` library using this http://bluebirdjs.com/docs/api/promisification.html
that way we dont have do the below promisification.

File: src/utils/node/pm2.ts [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+                return;
+            }
+            resolve();
+        })
+    })
+
+const startPM2 = async (installPath: string, name: string): Promise<void> =>
+    new Promise<void>((resolve, reject) => {
+        start({
+            name,
```
Comment (ManuGowda) [inline src/utils/node/pm2.ts]: I like your idea, but the downside of defining this pm2 config directly here will have issues when the core pm2 make changes, for example, the `app.js` thing or any other pm2 configuration changes. So, for every change in core we need to create a new release for the commander, for now, it should be fine, but let's discuss what would be the better approach for longer term.

File: src/utils/node/pm2.ts [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+            error: path.join(installPath, '/logs/lisk.app.err'),
+            log_date_format: 'YYYY-MM-DD HH:mm:ss SSS',
+            watch: false,
+            kill_timeout: 10000,
+            max_memory_restart: '1024M',
+            min_uptime: 20000,
+            max_restarts: 10,
+        }, err => {
+            if (err) {
+                reject(err);
```
Comment (ManuGowda) [inline src/utils/node/pm2.ts]: `return reject(err);` this should work, and same for `return resolve();`

**Review by ManuGowda — COMMENTED** [review]
@shuse2 I have reviewed the PR and it looks good, I left a few minor comments, some of them I am going to address, few comments are for your addition as well.

File: src/utils/node/pm2.ts (hunk 9 above)
Comment (shuse2) [reply src/utils/node/pm2.ts]: I think this is one of tslint rule that return type must match and explicit.

File: src/utils/node/pm2.ts (hunk 8 above)
Comment (shuse2) [reply src/utils/node/pm2.ts]: yea probably we should change the pm2 config to read from the core
we should move the pm2 config to the `core` =)

File: src/utils/node/pm2.ts (hunk 7 above)
Comment (shuse2) [reply src/utils/node/pm2.ts]: prefer not to use bluebird, and i tried `util` promisify, but it seems like typescript didnt like that...

Comment (ManuGowda) [reply src/utils/node/pm2.ts]: yeah I tried `util` promisify it's bit ugly, ok if its ok this way then fine.

File: src/utils/node/pm2.ts (hunk 9 above)
Comment (ManuGowda) [reply src/utils/node/pm2.ts]: ahhh ok 👍

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**